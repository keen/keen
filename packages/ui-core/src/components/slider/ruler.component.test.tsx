import React from 'react';
import { shallow } from 'enzyme';
import Ruler from './ruler.component';

describe('<Ruler />', () => {
  const props = {
    controlSize: 12,
    ticks: [
      {
        pos: 0,
        val: 0,
      },
      {
        pos: 10,
        val: 10,
      },
      {
        pos: 20,
        val: 20,
      },
    ],
  };

  it('should render correctly horizontal ruler', () => {
    const wrapper = shallow(<Ruler layout="horizontal" {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly vertical ruler', () => {
    const wrapper = shallow(<Ruler layout="vertical" {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
