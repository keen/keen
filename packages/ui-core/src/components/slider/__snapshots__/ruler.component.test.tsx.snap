// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ruler /> should render correctly horizontal ruler 1`] = `
<Fragment>
  <styled.div
    controlSize={12}
    key="0"
    left={0}
    type="horizontal"
  >
    <styled.div
      type="horizontal"
    />
    <styled.div
      type="horizontal"
    >
      0
    </styled.div>
  </styled.div>
  <styled.div
    controlSize={12}
    key="10"
    left={10}
    type="horizontal"
  >
    <styled.div
      type="horizontal"
    />
    <styled.div
      type="horizontal"
    >
      10
    </styled.div>
  </styled.div>
  <styled.div
    controlSize={12}
    key="20"
    left={20}
    type="horizontal"
  >
    <styled.div
      type="horizontal"
    />
    <styled.div
      type="horizontal"
    >
      20
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`<Ruler /> should render correctly vertical ruler 1`] = `
<Fragment>
  <styled.div
    controlSize={12}
    key="0"
    left={0}
    type="vertical"
  >
    <styled.div
      type="vertical"
    />
    <styled.div
      type="vertical"
    >
      0
    </styled.div>
  </styled.div>
  <styled.div
    controlSize={12}
    key="10"
    left={10}
    type="vertical"
  >
    <styled.div
      type="vertical"
    />
    <styled.div
      type="vertical"
    >
      10
    </styled.div>
  </styled.div>
  <styled.div
    controlSize={12}
    key="20"
    left={20}
    type="vertical"
  >
    <styled.div
      type="vertical"
    />
    <styled.div
      type="vertical"
    >
      20
    </styled.div>
  </styled.div>
</Fragment>
`;
